datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -231,7 +231,14 @@ impl FileScanConfig {
         for idx in proj_iter {
             if idx < self.file_schema.fields().len() {
                 let field = self.file_schema.field(idx);
-                table_fields.push(field.clone());
+
+                let f = if field.data_type().equals_datatype(&DataType::Binary) {
```

Comment:
If this works out I think we should put this feature behind a feature flag. I think this change will have the effect of changing all binary columns to utf8. This is the right thing to do for the `hits_partitioned` dataset, but I am not sure it is the right thing to do in general. 

I tried to write this up more coeherently in https://github.com/apache/datafusion/issues/12788